### PR TITLE
Fix lookupFactory deprecation

### LINKED ISSUE #644: DEPRECATION: Using "_lookupFactory" is deprecated.
When I am opening the ember inspector routes tab with Ember 2.12 I am seeing this warning:

schermafbeelding 2017-03-27 om 10 56 47

Probably [this line](https://github.com/emberjs/ember-inspector/blob/master/ember_debug/route-debug.js#L142) should use `factoryFor` when available?

### PR DESCRIPTION
**Pull request by rmachielse**
Fixes #644

### CODE REVIEW

File: ember_debug/route-debug.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ function buildSubTree(routeTree, route) {
 
       routeHandler = container.lookup('router:main').router.getHandler(handler);
       controllerName = routeHandler.get('controllerName') || routeHandler.get('routeName');
-      controllerFactory = container.lookupFactory(`controller:${controllerName}`);
+      controllerFactory = container.factoryFor ? container.factoryFor(`controller:${controllerName}`) : container.lookupFactory(`controller:${controllerName}`);
```
Comment (rwjblue) [inline ember_debug/route-debug.js]: Can you tweak this to use `owner.factoryFor` instead of `container.factoryFor`?

Comment (rmachielse) [inline ember_debug/route-debug.js]: @rwjblue 👍  I have changed that now

**Comment by mixonic** [comment]
I kicked the build. @rwjblue claims it is failing for an unrelated reason, and indeed the failure persists after rebuild.

I'd really like to see this get in soon regardless of Travis 😬

**Comment by mixonic** [comment]
That failure is reproducible locally and the same issues are reproducible on master.

I'm going to close this, and I'll also try to figure out the real failures in the next few days. I'd really, really like to have this fix released by Tuesday. 😁

**Comment by rmachielse** [comment]
@mixonic great, thanks!

**Comment by mixonic** [comment]
Opened a followup for the test issues on master: https://github.com/emberjs/ember-inspector/issues/648

**Pull request merged**